oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/MetricsQueryEsDAO.java

```diff
@@ -121,7 +134,43 @@ protected void functionAggregation(Function function, TermsAggregationBuilder pa
         return intValues;
     }
 
-    @Override public Thermodynamic getThermodynamic(String indName, Downsampling downsampling, List<String> ids, String valueCName) throws IOException {
+    @Override public IntValues[] getMultipleLinearIntValues(String indName, Downsampling downsampling,
+        List<String> ids, int numOfLinear, String valueCName) throws IOException {
+        String indexName = ModelName.build(downsampling, indName);
+
+        SearchResponse response = getClient().ids(indexName, ids.toArray(new String[0]));
+        Map<String, Map<String, Object>> idMap = toMap(response);
+
+        IntValues[] intValuesArray = new IntValues[numOfLinear];
+        for (int i = 0; i < intValuesArray.length; i++) {
+            intValuesArray[i] = new IntValues();
+        }
+
+        for (String id : ids) {
+            for (int i = 0; i < intValuesArray.length; i++) {
+                KVInt kvInt = new KVInt();
+                kvInt.setId(id);
+                kvInt.setValue(0);
+                intValuesArray[i].addKVInt(kvInt);
+            }
+
+            if (idMap.containsKey(id)) {
+                Map<String, Object> source = idMap.get(id);
+                IntKeyLongValueHashMap multipleValues = new IntKeyLongValueHashMap(5);
```

Comment:
I'm not very comfortable with the magic number here

Reply:
What do you expect? 5 should be suitable for 5-10 elements.

Reply:
> What do you expect? 5 should be suitable for 5-10 elements.

oops, didn't notice it is a map, I thought it was array mistakenly, ignore me 😄